### PR TITLE
fix: C SSHNPD memory leaks before SSH connection established

### PR DESCRIPTION
**Pull request by JeremyTubongbanua**
**- What I did**
- Pinned atsdk commit which is from https://github.com/atsign-foundation/at_c/pull/325
- Removed atlogger_free() since that is not the most recent feature

**- How to verify it**

valgrind leak summary after starting daemon, pkam authenticating, starting main loop, then Ctrl-C

```
==194277== LEAK SUMMARY:
==194277==    definitely lost: 0 bytes in 0 blocks
==194277==    indirectly lost: 0 bytes in 0 blocks
==194277==      possibly lost: 0 bytes in 0 blocks
==194277==    still reachable: 0 bytes in 0 blocks
==194277==         suppressed: 223 bytes in 1 blocks
```

### CODE REVIEW

**Comment by JeremyTubongbanua** [comment]
As of most recent commit of sending this message,

C SSHNPD works with at_c commit hash: 977cf8f51bb93e494b14fedfe3a577f440e45f0e

**Comment by JeremyTubongbanua** [comment]
Will wait for https://github.com/atsign-foundation/at_c/pull/325 to be pushed to trunk before we push this to trunk

**Review by xavierchanth — APPROVED** [review]

File: packages/c/sshnpd/sshnpd.supp [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+    fun:print
+    fun:cJSON_PrintUnformatted
+    fun:main
+}
+{
+  FIXME atclient_atsign_init
+    Memcheck:Leak
+    fun:malloc
+    fun:atclient_atsign_init
+}
```
Comment (xavierchanth) [inline packages/c/sshnpd/sshnpd.supp:16]: These also need to be unsuppressed & fixed to get rid of all pre-connection leaks

**Comment by JeremyTubongbanua** [comment]
Will do 1 more pass before I merge @XavierChanth

**Pull request merged**